draft-ietf-quic-transport.md

```diff
@@ -2420,6 +2420,12 @@ When a packet is detected as lost, the sender re-sends any frames as necessary:
 * STOP_SENDING frames MUST be retransmitted, unless the stream has become closed
   in the appropriate direction.  See {{solicited-state-transitions}}.
 
+* The most recent MAX_STREAM_DATA frame for a stream MUST be retransmitted. Any
+  previous unacknowledged MAX_STREAM_DATA frame for the same stream SHOULD NOT
+  be retransmitted since a newer MAX_STREAM_DATA frame for a stream obviates the
+  need for delivering older ones. Similarly, the most recent MAX_DATA frame MUST
+  be retransmitted; previous unacknowledged ones SHOULD NOT be retransmitted.
```

Comment:
I am starting to think it's a mistake to phrase this in terms of frames at all. It's the information you want to retransmit. It seems like there are lots of cases where you would be constantly updating your willingness to receive (if, say, you want to continue offering a given buffer size) but you only transmit occasionally. If I have to retransmit, why would I want to advertise stale data rather than the most current data.

Reply:
Agreed.  Anytime the words "retransmit frame" appear in the text, it's probably an error.  And the phrase 'retransmit packet' should occur twice: Once to tell you not to do it and once to tell you can do it for connection close packets only.